include/deal.II/distributed/shared_tria.h

```diff
@@ -111,6 +111,25 @@ namespace parallel
                                          const SubCellData &subcelldata);
 
       /**
+       * Copy @p old_tria to this triangulation.
+       *
+       * This function also partitions triangulation based on the MPI
+       * communicator provided to constructor.
+       */
+      virtual void copy_triangulation (const dealii::Triangulation<dim, spacedim> &old_tria);
```

Comment:
So this function really only makes sense if the argument is in fact a `p::s::Triangulation`. Any other class won't work. Why not just hardcode that type as the argument? Or do you want to do this because the function is `virtual` and you don't want to overload it?